mod/web_services/classes/Elgg/WebServices/Middleware/ViewtypeMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Elgg\WebServices\Middleware;
+
+use Elgg\Request;
+use Elgg\Exceptions\Http\BadRequestException;
+
+/**
+ * Set correct view_type for api calls
+ *
+ * @since 4.0
+ */
+class ViewtypeMiddleware {
+	
+	/**
+	 * Invoke middleware
+	 *
+	 * @param Request $request the Elgg request
+	 *
+	 * @return void
+	 * @throws BadRequestException
+	 */
+	public function __invoke(Request $request) {
+		
+		$viewtype = $request->getParam('view', 'json');
+		if (!ctype_alpha($viewtype)) {
```

Comment:
is ctype_alpha always available?

Reply:
yes and no. Enabled by default, can however be disabled. Not going to handle that